cmd/snap-update-ns/snapenv_test.go

```diff
@@ -0,0 +1,96 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main_test
+
+import (
+	"os"
+
+	. "gopkg.in/check.v1"
+
+	update "github.com/snapcore/snapd/cmd/snap-update-ns"
+)
+
+type envSuite struct{}
+
+var _ = Suite(&envSuite{})
+
+func (s *envSuite) TestSnapEnvHappy(c *C) {
+	restore := update.MockSnapConfineUserEnv("/run/user/1000/snap.snapname", "/home/user", "1000")
+	defer restore()
+	env, err := update.SnapEnv(1000)
+	c.Assert(err, IsNil)
+	c.Assert(env["XDG_RUNTIME_DIR"], Equals, "/run/user/1000/snap.snapname")
+	c.Assert(env["SNAP_REAL_HOME"], Equals, "/home/user")
+	c.Assert(env["SNAP_UID"], Equals, "1000")
+}
+
+// TODO: Find a way to test this
+func (s *envSuite) TestSnapEnvErrorOsEnv(c *C) {
+}
```

Comment:
You can mock the osutil call but I think in this case it's also ok to just remove this